exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -145,132 +167,234 @@ func (e *Exchange) SetDefaults() {
+	}
+	if err := e.Websocket.SetupNewConnection(&websocket.ConnectionSetup{
+		ResponseCheckTimeout:  exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:      exch.WebsocketResponseMaxLimit,
+		URL:                   wsSpotPrivate,
+		RateLimit:             request.NewWeightedRateLimitByDuration(2 * time.Millisecond),
+		Subscriber:            e.Subscribe,
+		Unsubscriber:          e.Unsubscribe,
+		GenerateSubscriptions: e.generatePrivateSubscriptions,
+		Handler:               e.wsHandleData,
+		Connector:             e.wsAuthConn,
+		MessageFilter:         connSpotPrivate,
+		Authenticate:          e.authenticateSpotAuthConn,
+	}); err != nil {
+		return err
+	}
+	wsFutures, err := e.API.Endpoints.GetURL(exchange.WebsocketFutures)
 	if err != nil {
 		return err
 	}
+	// Futures Public Connection
+	if err := e.Websocket.SetupNewConnection(&websocket.ConnectionSetup{
+		URL:                   wsFutures,
+		ResponseCheckTimeout:  exch.WebsocketResponseCheckTimeout,
+		RateLimit:             request.NewWeightedRateLimitByDuration(2 * time.Millisecond),
+		ResponseMaxLimit:      exch.WebsocketResponseMaxLimit,
+		Handler:               e.wsFuturesHandleData,
+		Subscriber:            e.SubscribeFutures,
+		Unsubscriber:          e.UnsubscribeFutures,
+		GenerateSubscriptions: e.generateFuturesSubscriptions,
+		Connector:             e.WsFuturesConnect,
+		MessageFilter:         connFuturesPublic,
+	}); err != nil {
+		return err
+	}
 
+	wsFuturesPrivate, err := e.API.Endpoints.GetURL(exchange.WebsocketFuturesPrivate)
+	if err != nil {
+		return err
+	}
+	// Futures Private Connection
 	return e.Websocket.SetupNewConnection(&websocket.ConnectionSetup{
-		ResponseCheckTimeout: exch.WebsocketResponseCheckTimeout,
-		ResponseMaxLimit:     exch.WebsocketResponseMaxLimit,
+		URL:                   wsFuturesPrivate,
+		ResponseCheckTimeout:  exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:      exch.WebsocketResponseMaxLimit,
+		RateLimit:             request.NewWeightedRateLimitByDuration(2 * time.Millisecond),
+		Handler:               e.wsFuturesHandleData,
+		Subscriber:            e.SubscribeFutures,
+		Unsubscriber:          e.UnsubscribeFutures,
+		GenerateSubscriptions: e.generateFuturesPrivateSubscriptions,
+		Connector:             e.futuresAuthConnect,
+		MessageFilter:         connFuturesPrivate,
+		Authenticate:          e.authenticateFuturesAuthConn,
+		Authenticated:         true,
 	})
 }
 
-// FetchTradablePairs returns a list of the exchanges tradable pairs
-func (e *Exchange) FetchTradablePairs(ctx context.Context, _ asset.Item) (currency.Pairs, error) {
-	// TODO: Upgrade to new API version for fetching operational pairs.
-	resp, err := e.GetTicker(ctx)
-	if err != nil {
-		return nil, err
-	}
+// FetchTradablePairs returns a list of the exchange's tradable pairs
+func (e *Exchange) FetchTradablePairs(ctx context.Context, assetType asset.Item) (currency.Pairs, error) {
+	switch assetType {
+	case asset.Spot, asset.Margin:
```

Comment:
You call this here, but nowhere else. Is margin supported? Does it need support with orderbooks and tickers and submitting orders etc?

Reply:
I would like an answer with this one. Has asset.Margin support has been forgotten to be added in this PR or should this case should be removed?